### ISSUE TITLE
CVC4 produces/complains about satisfiable unsat core

### ISSUE THREAD

**Issue by 4tXJ7f** [CVC4/CVC4#1953]
```
$ builds/bin/cvc4 --check-unsat-cores test/regress/regress0/uf/cnf_abc.smt2
(error "Internal error detected
void CVC4::SmtEngine::checkUnsatCore()
../../../../src/smt/smt_engine.cpp:5190:
SmtEngine::checkUnsatCore(): produced core was satisfiable."
```

**Comment by 4tXJ7f** [comment]
Note: `--check-proofs` succeeds (and the combination `--check-proofs --check-unsat-cores` also succeeds, which may indicate that CVC4 is not checking unsat cores when both options are enabled).

**Comment by 4tXJ7f** [comment]
My speculation above is not true. It looks like we are generating a different unsat core with proofs enabled (with 18 assertions instead of 14).

**Comment by 4tXJ7f** [comment]
Fixed by #1958.
